icu4c/source/common/dictbe.cpp

```diff
@@ -1090,14 +1106,31 @@ static inline bool isKatakana(UChar32 value) {
             (value >= 0xFF66 && value <= 0xFF9f);
 }
 
-
 // Function for accessing internal utext flags.
 //   Replicates an internal UText function.
 
 static inline int32_t utext_i32_flag(int32_t bitIndex) {
     return (int32_t)1 << bitIndex;
 }
 
+/** Mask constant for multiple UCharCategory bits (OP Open Punctuation). @stable ICU 2.1 */
+constexpr int32_t U_GC_OP_MASK = U_GC_PI_MASK | U_GC_PS_MASK;
+
+/** Mask constant for multiple UCharCategory bits (CP Close Punctuation). @stable ICU 2.1 */
+constexpr int32_t U_GC_CP_MASK =
+        U_GC_PD_MASK | U_GC_PE_MASK | U_GC_PC_MASK | U_GC_PO_MASK | U_GC_PF_MASK;
+
+static inline bool isOpenPunct(UChar32 c) {
```

Comment:
As I asked in the ICU4J review, should we use UnicodeSet here also the check the Punc mark instead? What is the benefit of using UTRIE2_GET16 and access propsTrie ?